### PR TITLE
WIP: Add long-term vision doc and basic info to README

### PR DESCRIPTION
**Pull request by tigrannajaryan**
I have been recently thinking about what should be the long-term vision for OpenTelemetry Agent/Collector. I believe it is important to have a clearly articulated set of high-level goals and guiding principles for the product that we work on. I took a first stab at it and would highly appreciate feedback on it and whether you think it is aligned with the goals of the OpenTelemetry community.

This has been already circulated internally at Omnition and also in Gitter channel https://gitter.im/open-telemetry/community

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
I signed it

**Comment by tigrannajaryan** [comment]
I signed it

**Comment by tigrannajaryan** [comment]
@pjanotti @bogdandrutu please have a look.

File: docs/VISION.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Observable
+Expose own operational metrics in a clear way. Be an exemplar of observable service. Allow configuring the level of observability (more or less metrics reported).
+
+## Multi-Data
+Support traces, metrics, logs and other relevant data types.
+
+## Usable Out of the Box
+Reasonable default configuration, supports popular protocols, runs and collects out of the box.
+
+## Extendible
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply docs/VISION.md]: Fixed.

File: docs/VISION.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+Extendible and customizable without touching the core code. Can create custom agents based on the core and extend with own components. Welcoming 3rd party contribution policy.
+
+## With Unified Codebase
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply docs/VISION.md]: Fixed.

File: docs/VISION.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The following are high-level items that define our long-term vision for OpenTelemetry Service, what we aspire to achieve. This vision is our daily guidance when we design new features and make changes to the Service.
+
+This is a living document that is expected to evolve over time.
+
+## Performant
+Highly stable and performant under varying loads. Well-behaved under extreme load, with predictable, low resource consumption.
+
+## Observable
+Expose own operational metrics in a clear way. Be an exemplar of observable service. Allow configuring the level of observability (more or less metrics reported).
```
[the remark this answers is not in the document]
Comment (tigrannajaryan) [reply docs/VISION.md]: Fixed.

File: docs/VISION.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# OpenTelemetry Service Long-term Vision
```
Comment (SergeyKanzhelev) [inline docs/VISION.md]: one point that needs to be covered is a relationship with FluentBit and Prometheus. Maybe in a separate document.

File: README.md [hunk 5]
```diff
@@ -1,2 +1,8 @@
-# opentelemetry service
-OpenTelemetry service
+# OpenTelemetry Service
+The OpenTelemetry service is a set of components that can collect traces, metrics and eventually other telemetry data (e.g. logs) from processes instrumented by OpenTelementry or other monitoring/tracing libraries (Jaeger, Prometheus, etc.), do aggregation and smart sampling, and export traces and metrics to one or more monitoring/tracing backends.
```
Comment (SergeyKanzhelev) [inline README.md]: also it does "enrichment" and in future transformation of telemetry. Like scrabbing of personal information or so...

File: README.md [hunk 6]
```diff
@@ -1,2 +1,8 @@
-# opentelemetry service
-OpenTelemetry service
+# OpenTelemetry Service
```
Comment (SergeyKanzhelev) [inline README.md]: great doc. Do you mind I will take pieces of it for the spec repo?

Comment (tigrannajaryan) [reply README.md]: I don't mind. Go ahead :-)

File: README.md (hunk 5 above)
Comment (tigrannajaryan) [reply README.md]: Sounds good, let me add that.

File: docs/VISION.md (hunk 4 above)
Comment (tigrannajaryan) [reply docs/VISION.md]: I am not familiar with this area, so perhaps someone else can add it separately (or I can do it later once I know more).

Comment (SergeyKanzhelev) [reply docs/VISION.md]: Created issue for the future: https://github.com/open-telemetry/opentelemetry-service/issues/10

File: docs/VISION.md [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+## Multi-Data
+Support traces, metrics, logs and other relevant data types.
+
+## Usable Out of the Box
+Reasonable default configuration, supports popular protocols, runs and collects out of the box.
+
+## Extensible
+Extensible and customizable without touching the core code. Can create custom agents based on the core and extend with own components. Welcoming 3rd party contribution policy.
+
+## Unified Codebase
```
Comment (songy23) [inline docs/VISION.md]: Suggestion: you can also mention unified configuration for Agent and Collector.

Comment (tigrannajaryan) [reply docs/VISION.md]: I tried to keep this doc very short and high-level only. The configuration unification is part of the general unification vision, so I think we can omit it in this high-level doc.

Comment (songy23) [reply docs/VISION.md]: Makes sense to me.

**Comment by tigrannajaryan** [comment]
@pjanotti please merge this.

**Pull request merged**